coupon_criteria_multi_product/models/coupon_criteria.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class CouponCriteria(models.Model):
+    _name = "coupon.criteria"
```

Comment:
You should do a migration script to rename the model. One simple example: https://github.com/OCA/stock-logistics-warehouse/blob/13.0/scrap_reason_code/migrations/13.0.1.1.0/pre-migration.py#L14

Reply:
Would this need to be done in all similar cases?

Reply:
If the model is created in your module and you change its name in a later iteration, you must do so. Otherwise you'll lose your previous data.